### PR TITLE
Update YAML config docs

### PR DESCRIPTION
**Pull request by mavam**
This is an addendum to #1059.

### CODE REVIEW

**Comment by mavam** [comment]
[ch19359]

File: doc/cli/vast.md [hunk 1]
```diff
@@ -32,13 +32,20 @@ for `vast start`.
...
-During startup, `vast` looks for a `vast.conf` in the current directory. If
-the file does not exist, `vast` then attempts to open `PREFIX/etc/vast.conf`
-where `PREFIX` is the installation prefix (which defaults to `/usr/local`).
+During startup, `vast` looks for a `vast.yaml` or `vast.yml` in the following
+directories, in order:
+
+1. `<sysconfdir>/vast`
+2. Environment-variables (stopping when first variable found):
+  1. `${XDG_CONFIG_HOME}/vast`
+  2. `${HOME}/.config/vast`
```
Comment (dominiklohmann) [inline doc/cli/vast.md:45]: I'd just rephrase this to say that we're respecting the XDG base directory specification for configuration files, with the default being `~/.config/vast/vast.yaml`.

File: doc/cli/vast.md [hunk 2]
```diff
@@ -32,13 +32,20 @@ for `vast start`.
...
-persisting option values and tweaking system parameters. Command line options
-always override configuration file values.
+In addition to command options, a YAML configuration file `vast.yaml` allows
+for persisting option values and tweaking system parameters. Command line
+options always override configuration file values.
 
-During startup, `vast` looks for a `vast.conf` in the current directory. If
-the file does not exist, `vast` then attempts to open `PREFIX/etc/vast.conf`
-where `PREFIX` is the installation prefix (which defaults to `/usr/local`).
+During startup, `vast` looks for a `vast.yaml` or `vast.yml` in the following
```
Comment (dominiklohmann) [inline doc/cli/vast.md:39]: Let's just not mention `yml` at all, but rather silently support it as a nicety. This allows you to specify full paths below, lowering the complexity for the reader.

**Review by dominiklohmann — CHANGES_REQUESTED** [review]

File: doc/cli/vast.md (hunk 1 above)
Comment (mavam) [reply doc/cli/vast.md:45]: I didn't know what that XDG thingy is unless you told me. Why not keep it explicit?

Comment (mavam) [reply doc/cli/vast.md:45]:
```diff
diff --git a/doc/cli/vast.md b/doc/cli/vast.md
index dbf94c2ce..065e86e55 100644
--- a/doc/cli/vast.md
+++ b/doc/cli/vast.md
@@ -36,8 +36,8 @@ In addition to command options, a YAML configuration file `vast.yaml` allows
 for persisting option values and tweaking system parameters. Command line
 options always override configuration file values.

-During startup, `vast` looks for a `vast.yaml` in the following directories, in
-order:
+VAST respects the XDG base directory specification for configuration files and
+looks for `vast.yaml` configuration at startup in the following order:

 1. `<sysconfdir>/vast`
 2. Environment-variables (stopping when first variable found):
```

Comment (dominiklohmann) [reply doc/cli/vast.md:45]: Because you're the odd one out here not knowing what the XDG base directory spec is.

I'd write the whole paragraph like this:

> During startup, VAST looks for configuration files in the following places, and merges their content with the more specific files taking a higher precedence:
> 
> 1. `<sysconfdir>/vast/vast.yaml`, where `<sysconfdir>` is the platform-specific directory for configuration files, e.g., `/etc/vast` for system-wide configuration.
> 2. `~/.config/vast/vast.yaml` for user-specific configuration. VAST respects the XDG base directory specification and its environment variables.
> 3. A configuration file passed using `--config=path/to/vast.yaml` on the command line for using a specific configuration file for individual commands.

Comment (mavam) [reply doc/cli/vast.md]: I'm glad I'm the odd one out by now. 😉 

I've pushed your changes with a slight simplification in the third bullet.

Comment (dominiklohmann) [reply doc/cli/vast.md]: LGTM!

**Review by dominiklohmann — APPROVED** [review]
